documentation/bolt_configuration_reference.md.erb

```diff
@@ -0,0 +1,155 @@
+
+## Log file configuration options
+
+You can configure how the results of plan runs are captured in a log file. Different log files can be configured by specifying the log file's location using either `console` or a path to the log file, such as `~/.puppetlabs/bolt/debug.log`.
+
+| Option | Description | Default |
+| ------ | ----------- | ------- |
+<% @log[:options].each do |option, desc| -%>
+| `<%= option %>` | <%= desc %> | <%= @log[:defaults][option].to_s %> |
+<% end %>
+
+### Example configuration
+
+Setting log configuration in the configuration file:
+
+```yaml
+log:
+  console:
+    level: error
+  ~/.puppetlabs/bolt/debug.log:
+    level: debug
+```
+
+## Plugin hooks configuration options
+
+The `plugin_hooks` section allows you to configure which plugins a specific hook should use for a specific target. This section is a hash where keys are hook names, and values specify and configure the plugin that that hook should use. There are two possible plugins: the `install_agent` plugin runs the `puppet_agent::install` task, and the `task` plugin runs your own custom task.
+
+For now, the only configurable plugin hook is `puppet_library`.
+
+The default is to use the `puppet_agent` plugin with the agent service stopped:
+
+```yaml
+plugin_hooks:
+  puppet_library:
+    plugin: puppet_agent
+    stop_service: true
+```
+
+The `puppet_agent::install` task will error if it's not run as root. To ensure this plugin to succeeds on non-root targets set `_run_as: true` for the `puppet_agent` plugin.
+
+```yaml
+plugin_hooks:
+  puppet_library:
+    plugin: puppet_agent
+    _run_as: root
+```
+
+You can use the `bootstrap` task to connect all targets to a PE master instead:
+
+```yaml
+plugin_hooks:
+  puppet_library:
+    plugin: task
+    _run_as: root
+    task: 'bootstrap'
+    parameters:
+      master: 'puppet.example.com'
+      cacert_content: <CERT>
+```
+
+You can also configure `plugin_hooks` using [`_plugin` references](https://puppet.com/docs/bolt/latest/using_plugins.html):
+
+```yaml
+plugin_hooks:
+  puppet_library:
+    plugin: puppet_agent
+    version:
+      _plugin: prompt
+      message: "Which version of Puppet do you want to install?"
+```
+
+## Puppetfile configuration options
+
+The `puppetfile` section of the configuration file configures how to retrieve modules when running `bolt puppetfile install`.
+
+| Option | Description |
+| ------ | ----------- |
+<% @puppetfile[:options].each do |option, desc| -%>
+| `<%= option %>` | <%= desc %> |
+<% end %>
+
+## Transport configuration options
+
+Transport configuration options can be set in both the configuration file and inventory file.
+
+<% @transports[:options].each do |transport, options| %>
+### <%= transport %>
+
+| Option | Description | Default |
+| ------ | ----------- | ------- |
+<% options.each do |option, desc| -%>
+| `<%= option %>` | <%= desc %> | <%= @transports[:defaults][transport][option].to_s || "" %> |
+<% end %>
+
+<% if transport == 'ssh' %>
```

Comment:
This just ensures that the OpenSSH options are listed right after ssh. Alternatively I could modify the options hash so ssh comes last and just list this at the end. Or if we don't really care about them being next to each other I can remove this check altogether and put it at the end of the page.